libbeat/metric/system/memory/memory.go

```diff
@@ -87,3 +87,48 @@ func AddSwapPercentage(s *SwapStat) {
 	perc := float64(s.Swap.Used) / float64(s.Swap.Total)
 	s.UsedPercent = common.Round(perc, common.DefaultDecimalPlacesCount)
 }
+
+// HugeTLBPagesStat includes metrics about huge pages usage
+type HugeTLBPagesStat struct {
+	sigar.HugeTLBPages
+	UsedPercent float64 `json:"used_p"`
+}
+
+// GetHugeTLBPages returns huge pages usage metrics
+func GetHugeTLBPages() (*HugeTLBPagesStat, error) {
+	pages := sigar.HugeTLBPages{}
+	err := pages.Get()
+
+	if err == nil {
+		return &HugeTLBPagesStat{HugeTLBPages: pages}, nil
+	}
+
+	if sigar.IsNotImplemented(err) {
+		return nil, nil
+	}
+
+	return nil, err
+}
+
+// AddHugeTLBPagesPercentage calculates ratio of used huge pages
+func AddHugeTLBPagesPercentage(s *HugeTLBPagesStat) {
+	if s.Total == 0 {
+		return
+	}
+
+	perc := float64(s.Total-s.Free+s.Reserved) / float64(s.Total)
+	s.UsedPercent = common.Round(perc, common.DefaultDecimalPlacesCount)
+}
+
+// GetHugeTLBPagesEvent returns the event created from huge page usage metrics
+func GetHugeTLBPagesEvent(s *HugeTLBPagesStat) common.MapStr {
```

Comment:
I think this function is not used?

Reply:
Umm, there are other similar `GetFooEvent` functions here, indeed they are not used in the beats repo, I though they were exported to be used somewhere else.

Reply:
@exekias WDYT? Should we keep these `GetFooEvent` functions?

Reply:
Removed new unused function, old ones are left there by now.